contracts/LightClientBridge.sol

```diff
@@ -0,0 +1,282 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/cryptography/ECDSA.sol";
+import "./utils/Bits.sol";
+import "./utils/Bitfield.sol";
+import "./ValidatorRegistry.sol";
+
+/**
+ * @title A entry contract for the Ethereum light client
+ */
+contract LightClientBridge {
+    using SafeMath for uint256;
+    using Bits for uint256;
+    using Bitfield for uint256[];
+
+    /* Events */
+
+    /**
+     * @notice Notifies an observer that the prover's attempt at initital
+     * verification was successful.
+     * @dev Note that the prover must wait until `n` blocks have been mined
+     * subsequent to the generation of this event before the 2nd tx can be sent
+     * @param prover The address of the calling prover
+     * @param blockNumber The blocknumber in which the initial validation
+     * succeeded
+     * @param id An identifier to provide disambiguation
+     */
+    event InitialVerificationSuccessful(address prover, uint256 blockNumber, uint256 id);
+
+    /**
+     * @notice Notifies an observer that the complete verification process has
+     *  finished successfuly and the new payload will be accepted
+     * @param prover The address of the successful prover
+     * @param payload the payload which was approved for inclusion
+     * @param id the identifier used
+     */
+    event FinalVerificationSuccessful(address prover, bytes32 payload, uint256 id);
+
+    /* Types */
+
+    struct ValidationData {
+        address senderAddress;
+        bytes32 payload;
+        uint256[] validatorClaimsBitfield;
+        uint256 blockNumber;
+    }
+
+    /* State */
+
+    ValidatorRegistry public validatorRegistry;
+    uint256 public currentId;
+    mapping(uint256 => ValidationData) public validationData;
+
+    /* Constants */
+
+    uint256 public constant THRESHOLD_NOM = 2;
+    uint256 public constant THRESHOLD_DENOM = 3;
```

Comment:
THRESHOLD_DENOMINOATOR right? i think put full word because denom is confusing in the bridge space as it can be used to refer to denomination, ie token symbol.